core/src/main/java/nu/yona/server/subscriptions/entities/UserPrivate.java

```diff
@@ -146,15 +144,15 @@ public UUID getNamedMessageSourceId()
 		return namedMessageSourceId;
 	}
 
-	public UUID getVpnLoginId()
-	{
-		// these are the same for performance
-		return userAnonymizedId;
-	}
-
-	public String getVpnPassword()
+	public Optional<String> takeVpnPassword()
```

Comment:
I think `getAndClearVpnPassword` is a bit clearer.

Reply:
Yes, I didn't like it very much either, but couldn't find this simple alternative.